pkg/api/v1/resources/resource_interface.go

```diff
@@ -51,6 +51,8 @@ type InputResource interface {
 	SetStatus(status *core.Status)
 	GetNamespacedStatuses() *core.NamespacedStatuses
 	SetNamespacedStatuses(namespacedStatuses *core.NamespacedStatuses)
```

Comment:
can we perhaps embed the messages in the pre-existing details?

```proto
message Status {
    enum State {
        // Pending status indicates the resource has not yet been validated
        Pending = 0;
        // Accepted indicates the resource has been validated
        Accepted = 1;
        // Rejected indicates an invalid configuration by the user
        // Rejected resources may be propagated to the xDS server depending on their severity
        Rejected = 2;
        // Warning indicates a partially invalid configuration by the user
        // Resources with Warnings may be partially accepted by a controller, depending on the implementation
        Warning = 3;
    }
    // State is the enum indicating the state of the resource
    State state = 1;
    // Reason is a description of the error for Rejected resources. If the resource is pending or accepted, this field will be empty
    string reason = 2;
    // Reference to the reporter who wrote this status
    string reported_by = 3;
    // Reference to statuses (by resource-ref string: "Kind.Namespace.Name") of subresources of the parent resource
    map<string, Status> subresource_statuses = 4;

    // Opaque details about status results
    google.protobuf.Struct details = 5;
}
```

a namespaced status already supports this opaque metadata

this is better since it will play nicely with our multi-controller status reporter, rather than requiring you to reimplement

Reply:
Added!